components/fxa-client/src/ffi.rs

```diff
@@ -120,35 +123,17 @@ impl From<AccessTokenInfo> for AccessTokenInfoC {
     }
 }
 
-#[repr(C)]
-pub struct ProfileC {
-    uid: *mut c_char,
-    email: *mut c_char,
-    avatar: *mut c_char,
-    avatar_default: u8, // JNA dislikes booleans.
-    display_name: *mut c_char,
-}
-
-impl Drop for ProfileC {
-    fn drop(&mut self) {
-        unsafe {
-            destroy_c_string(self.uid);
-            destroy_c_string(self.email);
-            destroy_c_string(self.avatar);
-            destroy_c_string(self.display_name);
-        }
-    }
-}
-
-impl From<Profile> for ProfileC {
-    fn from(profile: Profile) -> Self {
-        ProfileC {
-            uid: rust_string_to_c(profile.uid),
-            email: rust_string_to_c(profile.email),
-            avatar: rust_string_to_c(profile.avatar),
-            avatar_default: profile.avatar_default as u8,
-            display_name: opt_rust_string_to_c(profile.display_name),
+impl From<Profile> for ffi_types::Profile {
+    fn from(p: Profile) -> Self {
+        let mut profile = ffi_types::Profile::new();
```

Comment:
I don't like this part, as for every field added we add to a struct passed over to the FFI we'll must not forget to use the setter.

Reply:
[Prost](https://github.com/danburkert/prost) has a better API in my opinion, but I'm not quite sure how well it's maintained.

Reply:
tower-grpc uses/recommends Prost, and the consulting firm to build our GCP API's was using it. I'd consider it a better option given those endorsements.